scripts/setup.sh

```
@@ -0,0 +1,236 @@
+#! /bin/bash
+
+# This script is used to setup the environment for the project
+
+set -euo pipefail
+
+function version_compare() {
+    local version1=$1
+    local version2=$2
+    local IFS=.
+    local i ver1=($version1) ver2=($version2)
```

Shellcheck (0.10.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)